### DISCUSSION TITLE
Which is the official url of an event when it appears on multiple sites?

### DISCUSSION THREAD

**Discussion by saumier** [culturecreates/artsdata-data-model#112] [Data Model]
In Artsdata, when the same event is on multiple website, there can be several urls for the event.   This data is stored in Artsdata. But when a consumer of data in Artsdata wants to reuse the event on their listing site, how should Artsdata decide which is the official url for the event?

In other words, on a 3rd party listing site that displays events from Artsdata in a list without event detail pages, where should the human user be taken if they click on the event in the list?  Or should there be a list of urls to choose from?

N.B. The "buy" url is usually the ticketing platform and there is only one ticketing platform for each event so there is no question about the "buy" url.

N.B. This question needs some good User Stories to guide us.

**Comment by tammy-culture** [comment]
My assumption/reaction is that if we had only 1 choice, I would display the buy link.  My 2 cents.

**Reply by troughc** [reply]
I agree with Tammy

**Reply by MaudeFJ** [reply]
I might have a different opinion.. let's say some web sites are on arts data because Signé Laval paid for their connection so they could be in the Signé Laval calendar, maybe it could be fair that the website displayed on other websites that use arts data  data is Signé Laval's website, but also from an user perspective the buy link is more logical.

**Reply by troughc** [reply]
@MaudeFJ You raise a very good point as it is important to promote Signe Laval / generate interest and traffic to your website. I have a few questions and/or ideas to add to a discussion. 

First, what is the mandate of Signe Laval? To my mind this is an important consideration in this discussion. Is it for people to discover the events you are promoting and sources of those events? Is it to generate visits your site? Is it more than one thing that is equally important?

If an overarching mission to promote the presenters and there events, what is the most effective way to do this (in terms of where to send a person discovering an event on the web)?

I am certain it is also important to promote your contributions to the overall success of event promotion - so that you continue to be funded and can continue to provide this value to the community. With that in mind, maybe we should brainstorm on ways to highlight the contributions of Signe Laval (and other sites)  whilst also sending visitors to the buy link (so they don't have to go through multiple sites to get to the event).

**Reply by MaudeFJ** [reply]
Totally! My first mandate is to promote Laval arts and culture, and I'm doing so by being relevant (so people click on my site) = both are true!  I feel a brainstorm would be pertinent and also that the user flow straight to the ticketing is better.

**Reply by saumier** [reply]
@tammy-culture My question was more about which url to use for the event when there are more than one. The "buy" url is passed in the offer.  I updated the original question by adding more explanation about the "buy" url and that it is not about that.  I also realized that none of the listing sites so far like LaVitrine, SignéLaval, Tout Culture or D'ici have a "more info" button so that was a bad user story on my part. Is the question more clear?

**Reply by fjjulien** [reply]
I disagree with Tammy's suggestion to give priority to the buy link. 

Two reasons: 

1. A buy link can point to a reseller. That would be really bad if, even once, Arsdata blindly recommended a resale site because of an automated ranking method.
2. The official event organizer should always have priority (see my comment below).

**Comment by fjjulien** [comment]
Lorsqu'il existe plusieurs URL pour un même événement, je crois qu'Artsdata devrait toujours privilégier l'URL de la page d'énénement sur le site de l'organisateur officielle (par exemple, le diffuseur).

Pourquoi? 

Parce qu'il s'agit de la source qui a l'autorité sur l'événement : la source qui a acquis les droits pour présenter la représentation. C'est à cette source que l'on peut faire confiance pour déterminer quel est le bon lien pour la billetterie (et ainsi éviter de tomber sur un site de revente ou, pire, un site de fraude). C'est cette source qui devrait normalement être la plus à jour et la plus fiable. C'est la source où on peut espérer trouver un bon vieux numéro de téléphone si on a une question.  et que c'est sur elle que toutes les autres sources s'appuient pour obtenir leur données et contenus.

Comment peut-on faire pour déterminer quel site est celui de l'organisateur officiel?

1. Dans les données structurées Schema, sous la propriété `organizer`.
2. Si l'entitée imbriquée sous `organizer` n'a pas d'identifiant `sameAs` mais a un `url`, on peut utiliser un regex pour comparer le domaine de l'organisateur et le domaine de la page d'événement.
3. En cas de conflit dans les données `organizer` (deux sites indiquent des valeurs différentes) mais que les entités `organizer` sont désambiguïsables, on pourrait vérifier si l'une des entités est un diffuseur (à partir des données importées de RIDEAU, de CAPACOA, de Scènes francophones, auquel on a ajouté le `additionalType : adr:PresentingOrganization`) et présumer que le diffuseur est l'organisateur officiel. Ce sera vrai dans 99% des cas.
4. Si on veut s'éviter la tâche d'analyser les contenus des données imbriquées sous `organizer` (et sachant qu'elles sont souvent absentes). On pourrait aussi avoir des règles pour identifier des "trusted domains" : tout domaine associé à un organisme membre d'un réseau de diffusion et/ou avec `additionalType : adr:PresentingOrganization` pourrait être considéré comme un domaine de confiance pour une page d'événement.

Note en regard du point 4 : Il arrive fréquemment que des diffuseurs gestionnaires de salle louent leur salles et offrent en même temps le service d'afficher une page web d'événement sur leur site pour leur client. Dans ces-cas, il est aussi possible que le client aient sa propre page d'événement sur son site. Cela dit, même dans ces circonstances, le site du diffuseur/gestionnaire de salle demeurerait une source fiable.

**Comment by fjjulien** [comment]
On complément à la suggestion précédente, on pourrait aussi envisager de catégoriser des URLs selon qu'ils sont des acteurs de la chaîne du spectacle (diffuseur, producteur, artiste), des agrégateurs, des billetteries fiables, des sites de revente, puis accorder un classement à chaque groupe. Christian Roy a déjà employé une telle méthode pour le projet de mesure de la découvrabilité.

Cette classement pourrait être en partie généré par des requêtes, mais il devrait être modifiable par des humains. Ça prendrait une fonction permettant à un usager de soumettre une URL avec une catégorisation et une annotation justifiant sa catégorisation. Ça peut sembler une démarche excessivement onéreuse en terme de ressources, mais les bénéfices pourraient être significatifs pour les consommateurs de données. Dans le long terme, ça pourrait même devenir un produit à valeur ajoutée.

**Comment by fjjulien** [comment]
Voici une autre idée...

En plus des autres méthodes proposées plus haut, le classement des URL pourrait aussi s'appuyer sur une analyse des données structurées sur le site web. En cas d'hésitation entre deux pages web similaires, on devrait privilégier celle qui a les données structurées d'événement les plus riches. 

Qu'est que j'entends par « données structurées riches »? Une page web qui a : 

1. Toutes les propriétés exigées par Artsdata;
2. Plusieurs propriétés recommandées par Artsdata;
3. Des liens sameAs vers bases de connaissances ouvertes (Artsdata, Wikidata) - des liens externes vers des bases de connaissances contribuent à renforcer la découvrabilité de tout l'écosystème et les pages qui en comportent devraient donc être privilégiés.

Pour y arriver, il faudrait développer un script pour analyser les données structurées et générer un pointage. Pour cela, on pourrait s'appuyer sur le [chiffrier de statitistiques de données structurées](https://docs.google.com/spreadsheets/d/1j258cga1dOs_vd7LxwkyutF0RpOIH6AFoCPu8YeqazE/edit?usp=sharing) développé par @tammy-culture, @dlh28 et moi.

Cette méthode aurait comme avantage de faire ressortir du lot les sites alimentés par le CMS Footlight (dont Culture Laval) et d'autres sites comme Scènes francophones qui sont des sites que l'on peut qualifier d'aggrégateur, mais dont les données sont riches et fiables.

**Comment by fjjulien** [comment]
Je n'ai pas d'autre idée. Pour le moment... ;)

**Comment by saumier** [comment]
Here is the list of the current properties in Artsdata used to store urls for an Event:
1. event.url  --> this is currently where the official url of the event should go
2. event.wasDerivedFrom --> this is the URI that was used to derive the data. It is automatically added by the crawler when the data comes from raw json-ld on a webpage and ensures there is a path back to the source webpage.
3.  event.mainEntityofPage -->  this links to the WebPage entity about this event but not necessarily the official page. There can be multiple webpages with properties such as English and French allowing more flexibility to direct the user.
